waiter/src/waiter/core.clj

```diff
@@ -1485,6 +1490,22 @@
                              (wrap-secure-request-fn
                                (fn waiter-auth-handler-fn [request]
                                  {:body (str (:authorization/user request)), :status 200})))
+   :waiter-auth-saml-acs-handler-fn (pc/fnk [[:state authenticator]]
+                                      (let [{:keys [saml-acs-handler-fn]} authenticator]
+                                        (fn waiter-auth-saml-acs-handler-fn [request]
+                                          (when-not saml-acs-handler-fn
+                                            (throw (ex-info "Current authenticator can not respond to SAML assertion message"
+                                                            {:authenticator authenticator
+                                                             :status 400})))
+                                          (saml-acs-handler-fn request authenticator))))
+   :waiter-auth-saml-auth-redirect-handler-fn (pc/fnk [[:state authenticator]]
+                                      (let [{:keys [saml-auth-redirect-handler-fn]} authenticator]
+                                        (fn waiter-auth-saml-auth-redirect-handler-fn [request]
+                                          (when-not saml-auth-redirect-handler-fn
```

Comment:
I don't think `saml-auth-redirect-handler-fn` should come from the authenticator via field access. Why not refer to it directly?

Reply:
done